### PR TITLE
Add jcl as logging facade

### LINKED ISSUE #267: Define a logging strategy
In several classes, we can find statements such as:
System.out.println("something");
ex.printStackTrace();

These statements should be replaced with some logging statements such as
LOG.info("something");
LOG.error(ex);

Basically, in java we have 3 different logging facades:
- java.util.logging (jul)
- jakarta commons logging (jcl)
- simple logging for java (slf4j)

- jul has the advantage of being included in java core library. Unfortunately, most developers think this api is useless and not very flexible.
- jcl is, by far, one of the most popular logging library. It is already a dependency of ews-java-api! Its main drawback is its rather old API that does not leverage java 5+ features (such as var args).
- slf4j is now very popular as well. It provides several bridges so that it can be used with almost any other logging facades (such as jcl for example). It leverages java 5+. So instead of writing:
```
if(LOG.isDebugEnabled()) {
  LOG.debug("foo "+bar);
}
```
we can directly write:
```
LOG.debug("foo {}", bar);
```
In the first snippet, since the arguments are evaluated before executing the method, the runtime is generating a string that would never be used if the logging level is higher than debug. To avoid such overhead, developers have no choice but to add this ugly if statement.
In the second snippet, leveraging var args, no need to add an if statement, the string will be created only if the log level is debug.

I opened this issue so that we can decide if we want to use a logging framework and if yes, which one we want to use.
My reference goes to slf4j for its efficience & flexibility.
Any thoughts ?
(Of course I'm ready to do a PR for this issue)

### PR DESCRIPTION
**Pull request by Nowheresly**
logback and jansi are added for test purpose only
all System.out.print are replaced with regular LOG.debug statements
all e.printStackTrace() are replaced with regular LOG.error statements
Fixes #267 
This is the same PR as #275 , but this time, commons-logging is used as logging facade.

Of course, either this one or #275 can be merged, but not both.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@Nowheresly__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by serious6** [comment]
> tagged for 2.0 as mentioned in [comment](https://github.com/OfficeDev/ews-java-api/pull/275#issuecomment-89995721)

**Comment by serious6** [comment]
@Nowheresly once upgraded, ready to merge.

**Comment by Nowheresly** [comment]
done

File: src/test/resources/logback-test.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<configuration scan="true" debug="true">
```
Comment (serious6) [inline src/test/resources/logback-test.xml]: can you also add project license-header here?

**Comment by serious6** [comment]
@Nowheresly thanks. After you added the missing license-header we are ready :stuck_out_tongue_winking_eye:

**Comment by serious6** [comment]
@Nowheresly thanks for your contribution

**Pull request merged**